test/test_lazy.py

```diff
@@ -0,0 +1,188 @@
+#         DTAG_BOUNDARY("y"): NeumannDiffusionBoundary(0.)
+#     }
+#
+#     def op(alpha, u):
+#         return _gradient_operator(
+#             discr, DISCR_TAG_BASE, alpha, boundaries, u)
+
+#     compiled_op = actx.compile(op)
+#     alpha = discr.zeros(actx) + 1
+#     u = discr.zeros(actx)
+#     compiled_op(alpha, u)
+
+
+# FIXME: Re-enable and fix up if/when standalone divergence operator exists
+# def test_lazy_op_divergence(ctx_factory):
+#     cl_ctx = ctx_factory()
+#     actx, discr = _op_test_fixture(cl_ctx)
+#
+#     from grudge.dof_desc import DTAG_BOUNDARY, DISCR_TAG_BASE
+#     from mirgecom.diffusion import (
+#         _divergence_alpha_operator,
+#         DirichletDiffusionBoundary,
+#         NeumannDiffusionBoundary)
+#
+#     boundaries = {
+#         DTAG_BOUNDARY("x"): DirichletDiffusionBoundary(0.),
+#         DTAG_BOUNDARY("y"): NeumannDiffusionBoundary(0.)
+#     }
+#
+#     def op(alpha, u):
+#         return _divergence_alpha_operator(
+#             discr, DISCR_TAG_BASE, alpha, boundaries, u)
+
+#     compiled_op = actx.compile(op)
+#     alpha = discr.zeros(actx) + 1
+#     u = make_obj_array([discr.zeros(actx) for _ in range(2)])
+#     compiled_op(alpha, u)
+
+
+def test_lazy_op_diffusion(ctx_factory):
+    cl_ctx = ctx_factory()
+    actx, discr = _op_test_fixture(cl_ctx)
+
+    from grudge.dof_desc import DTAG_BOUNDARY, DISCR_TAG_BASE
+    from mirgecom.diffusion import (
+        diffusion_operator,
+        DirichletDiffusionBoundary,
+        NeumannDiffusionBoundary)
+
+    boundaries = {
+        DTAG_BOUNDARY("x"): DirichletDiffusionBoundary(0.),
+        DTAG_BOUNDARY("y"): NeumannDiffusionBoundary(0.)
+    }
+
+    def op(alpha, u):
+        return diffusion_operator(
+            discr, DISCR_TAG_BASE, alpha, boundaries, u)
+
+    compiled_op = actx.compile(op)
+
+    alpha = discr.zeros(actx) + 1
+    u = discr.zeros(actx)
+
+    eager_result = op(alpha, u)
+    lazy_result = compiled_op(alpha, u)
+    assert discr.norm(lazy_result - eager_result, np.inf) < 1e-15
+
+
+def test_lazy_op_euler(ctx_factory):
+    cl_ctx = ctx_factory()
+    actx, discr = _op_test_fixture(cl_ctx)
+
+    from grudge.dof_desc import DTAG_BOUNDARY
+    from mirgecom.fluid import make_conserved
+    from mirgecom.eos import IdealSingleGas
+    from mirgecom.boundary import AdiabaticSlipBoundary
+    from mirgecom.euler import euler_operator
+
+    eos = IdealSingleGas()
+
+    boundaries = {
+        DTAG_BOUNDARY("x"): AdiabaticSlipBoundary(),
+        DTAG_BOUNDARY("y"): AdiabaticSlipBoundary()
+    }
+
+    def op(cv):
+        return euler_operator(discr, eos, boundaries, cv)
+
+    compiled_op = actx.compile(op)
+
+    cv = make_conserved(
+        dim=2,
+        mass=discr.zeros(actx) + 1,
+        energy=discr.zeros(actx) + 1,
+        momentum=discr.zeros(actx) + np.array([1, 2]),
+        species_mass=discr.zeros(actx) + np.array([0.5, 0.25, 0.25]))
```

Comment:
@MTCam Is there an easy way to make this physical? I just need to change the energy, right? Or does `euler_operator` not care?

Reply:
Euler op won't care, but if you set `energy = 2.5 + .5*np.dot(momentum, momentum)/mass` it'll get P=1.0.